Use BaseStore registry to get data types and data type classes

brendan-m-murphy opened the pull request:
* **Summary of changes** (Bug fix, feature, docs update, ...)

Updated `define_data_type_classes` to return the subclass registry from `BaseStore`.

Then `define_data_types` just returns a tuple of keys from the data type classes dictionary.


* **Please check if the PR fulfills these requirements**

- [ ] Closes #xxxx (Replace xxxx with the Github issue number)
- [ ] [Tests added and passed](https://docs.openghg.org/development/python_devel.html#testing) if fixing a bug or adding a new feature
- [ ] All code checks passing - `black --line-length 110` run over code, `mypy` and `flake8` not showing any errors
- [ ] Documentation and tutorials updated/added
- [ ] Added an entry in the latest `CHANGELOG.md` file if fixing a bug or adding a new feature
- [ ] Added any new requirements to `requirements.txt` and `recipes/meta.yaml`

rt17603 commented:
One thing to note, if I look at the `_registry` attribute we have a bit of a weird dict because we have a `MetStore` class which doesn't include a `_data_type` attribute.
```
In [3]: BaseStore._registry
Out[3]: 
{'boundary_conditions': openghg.store._boundary_conditions.BoundaryConditions,
 'flux': openghg.store._flux.Flux,
 'eulerian_model': openghg.store._eulerian_model.EulerianModel,
 'footprints': openghg.store._footprints.Footprints,
 'mobile': openghg.store._obsmobile.ObsMobile,
 'column': openghg.store._obscolumn.ObsColumn,
 'surface': openghg.store._obssurface.ObsSurface,
 '': openghg.store._metstore.METStore,
 'flux_timeseries': openghg.store._flux_timeseries.FluxTimeseries}
```

The [`met_v5_add_data`](https://github.com/openghg/openghg/blob/met_v5_add_to_store/openghg/store/_met.py) branch actually adds a new data type class (`_met.py`) which I believe should supercede this anyway so perhaps we could just delete `_metstore.py` (or, short term, add a `_data_type` attribute) to make sure this dict doesn't include an empty string.

brendan-m-murphy commented:
> One thing to note, if I look at the `_registry` attribute is that we have a bit of a weird dict because we have a `MetStore` class which doesn't include a `_data_type` attribute.
> 
> ```
> In [3]: BaseStore._registry
> Out[3]: 
> {'boundary_conditions': openghg.store._boundary_conditions.BoundaryConditions,
>  'flux': openghg.store._flux.Flux,
>  'eulerian_model': openghg.store._eulerian_model.EulerianModel,
>  'footprints': openghg.store._footprints.Footprints,
>  'mobile': openghg.store._obsmobile.ObsMobile,
>  'column': openghg.store._obscolumn.ObsColumn,
>  'surface': openghg.store._obssurface.ObsSurface,
>  '': openghg.store._metstore.METStore,
>  'flux_timeseries': openghg.store._flux_timeseries.FluxTimeseries}
> ```
> 
> The [`met_v5_add_data`](https://github.com/openghg/openghg/blob/met_v5_add_to_store/openghg/store/_met.py) branch actually adds a new data type class (`_met.py`) which I believe should supercede `MetStore` anyway so perhaps we could just delete `_metstore.py` (or, short term, add a `_data_type` attribute) to make sure this dict doesn't include an empty string.

Yea the registry only works if each class sets their `_data_type` attribute to something besides the value `''` inherited from `BaseStore`. I can try to make it do a sensible default.

brendan-m-murphy commented:
I added a check to make sure subclasses of `BaseStore` set a non-trivial value for `_data_type`. I added a `_data_type` attribute to `METStore` since otherwise the error is raised and the tests won't run when `METStore` is imported.

rt17603 commented:
> > Can we just make sure the tick boxes are completed for this? Just makes it easier to evaluate that the steps have been done / not needed.
> 
> Sorry, I wasn't sure exactly what we want for a very small PR... I can open an issue and link it.

No problem and that's fine - we don't always need an issue - just mainly trying to make sure our PRs are consistent so we have a good record for this and to make sure we know what's been done when reviewing.

rt17603 reviewed (approved):
Looks good!
